TVC_MVP/CountyViewController.swift

```diff
@@ -38,37 +40,38 @@ class CountyViewController: UIViewController, UITableViewDelegate, UITableViewDa
     }
 
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-          return presenter.numberOfRowsInSection()
+        return presenter.numberOfRowsInSection()
     }
 
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let county = presenter.fetchCounty(for: indexPath) else { fatalError(Constants.errorText) }
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.cellIdentifier, for: indexPath) as? CountyTableViewCell else { return UITableViewCell() }
+        guard let county = presenter.getObject(at: indexPath) else { return UITableViewCell() }
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: Constants.cellIdentifier,
+                                                       for: indexPath) as? CountyTableViewCell else { return UITableViewCell() }
         cell.setup(with: county)
         return cell
     }
 
     // MARK: - UITableViewDelegate methods
 
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        presenter.presentAlert(at: indexPath)
+        presenter.didSelectRow(indexPath: indexPath)
     }
 
     // MARK: - CountyViewProtocol methods
 
-    func presentAlert(title: String?, message: String?, actions: AlertAction...) {
-        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
-        actions.forEach { action in
-            alert.addAction(UIAlertAction(title: action.title, style: (action.style?.mapStyle)!, handler: { _ in
-                action.action?()
-            }))
-        }
-        present(alert, animated: true, completion: nil)
-    }
-
     func reloadData() {
         tableView.reloadData()
     }
 
+    func presentAlert(title: String, message: String, action: (() -> Void)?) {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        guard let action = action else { return }
```

Comment:
The `action` is optional, but if we don't pass it, then the alert won't be presented? When using the optionality we should think what will happen with different inputs, and what would be the expected result. If we don't want to present the alert when there is no action, then the action shouldn't be optional.